### DISCUSSION TITLE
mcast_group_addr is not taking the input value

### DISCUSSION THREAD

**Discussion by cyanaryan** [belaban/JGroups#770] [General]
I am working on upgrading the jgroup to 4.x version. So while updating the configuration file I am currently doing this

```
<config xmlns="urn:org:jgroups"
        xmlns:xsi="[http://www.w3.org/2001/XMLSchema-instance"](http://www.w3.org/2001/XMLSchema-instance%22)
        xsi:schemaLocation="urn:org:jgroups [http://www.jgroups.org/schema/jgroups-4.2.xsd"](http://www.jgroups.org/schema/jgroups-4.2.xsd%22)>
   <UDP
         mcast_addr="${cluster.udp.mcast_addr:FF05::2592}"
         mcast_port="${cluster.udp.mcast_port:46654}"
         tos="8"
         ucast_recv_buf_size="20000000"
         ucast_send_buf_size="640000"
         mcast_recv_buf_size="25000000"
         mcast_send_buf_size="640000"
         max_bundle_size="64000"
         ip_ttl="${jgroups.udp.ip_ttl:2}"
         enable_diagnostics="false"
 
         thread_naming_pattern="pl"
 
         thread_pool.enabled="true"
         thread_pool.min_threads="2"
         thread_pool.max_threads="30"
         thread_pool.keep_alive_time="20000"
         />
 
   <PING/>
   <MERGE3 max_interval="30000" min_interval="10000"/>
   <FD_SOCK/>
   <FD_ALL/>
   <BARRIER flush_timeout="30000" />
   <pbcast.NAKACK2   
                  xmit_table_max_compaction_time="30000"             
                   use_mcast_xmit="true"                 
                   discard_delivered_msgs="true"/>
   <UNICAST3 conn_close_timeout="0"/>
   <pbcast.STABLE stability_delay="1000" desired_avg_gossip="50000" max_bytes="1000000"/>
   <pbcast.GMS print_local_addr="false" join_timeout="3000" view_bundling="true"/>
   <UFC max_credits="500000" min_threshold="0.20"/>
   <MFC max_credits="500000" min_threshold="0.20"/>
   <FRAG2 frag_size="60000"  />
   <!-- <pbcast.STREAMING_STATE_TRANSFER/> -->
   <pbcast.FLUSH timeout="0"/>
   <pbcast.STATE_SOCK pool_thread_keep_alive="30000"/>
</config>
```
Currently, the value in the mcast_group_addr it's taking 228.8.8.8 and not the ff05. In the previous version I was getting the ff05 address. Can u please specify, why is that so? Can u please help me on how to correct that.

Previous version: 2.x
Previous mcast_group_addr=/ff05:0:0:0:0:0:0:4

**Comment by belaban** [comment]
I'll take a look. It might be incorrectly parsed because of the multiple colons (`::`) present in the IPv6 address...

**Comment by rhusar** [comment]
In case this is really parsed incorrectly (the correct parsing IIUC should be find *first* colon and everything before is the expression and everything after is the value) you can workaround as:

1. remove the expression and just put in the value  i.e. `mcast_addr="FF05::2592"`
2. remove the value and pass it via system property - i.e. `mcast_addr="${cluster.udp.mcast_addr}"` and start process with `-Dcluster.udp.mcast_addr=FF05::2592`

**Reply by cyanaryan** [reply]
@rhusar 
I am using the first approach `mcast_addr="FF05::2592"` .Still it is taking the default value 228.8.8.8

**Comment by belaban** [comment]
The reason `mcast_addr` is not used is that you're using an IPv4 stack: since `bind_addr` is null and `-Djava.net.preferIPv6Addresses=true` is not set, JGroups prefers IPv4 addresses. This means that the value of `mcast_addr` is ignored [1].

You can change this by either setting `-Djava.net.preferIPv4Stack=false -Djava.net.preferIPv6Addresses=true` or by setting `bind_addr` in the configuration file to an IPv6 value.

[1] http://www.jgroups.org/manual5/index.html#IPAddresses

**Comment by belaban** [comment]
The parsing is actually correct
